### DISCUSSION TITLE
Is there a complete app descriptor documentation?

### DISCUSSION THREAD

**Discussion by PippoApps** [airsdk/Adobe-Runtime-Support#2411] [General]
I often find myself looking for bits and pieces of tags for app descriptor and relying on my notes.
Is there some official up to date documentation available?
I had found some old stuff from Adobe but very outdated.
thx

**Comment by ajwfrost** [comment]
We're trying to maintain this now under https://airsdk.dev/docs/building/application-descriptor-files/introduction
Please let us know if you see any omissions!

thanks

**Reply by PippoApps** [reply]
Thank @ajwfrost, this is very useful

**Comment by PippoApps** [comment]
@ajwfrost I just noticed this is missing in initialWindow:
`<requestedDisplayResolution>high</requestedDisplayResolution>`

**Reply by ajwfrost** [reply]
Thanks @PippoApps  - apologies for the delay, we've just gone through and updated this and a few other ones where our recent changes hadn't been reflected in that documentation
https://airsdk.dev/docs/building/application-descriptor-files/elements/initialWindow#requesteddisplayresolution

thanks

**Reply by FliplineStudios** [reply]
Quick side note: that requestedDisplayResolution feature only works on Macs, correct?  The original Adobe documentation mentioned it was only for "MacBook Pro" computers, and I've never been able to get it to work on Windows PCs with 4K monitors.  I don't remember seeing anything in release notes ever mentioning adding Windows support, unless it was added at some point and I missed it!

**Reply by PippoApps** [reply]
Thank you @ajwfrost.
@FliplineStudios according to my tests it works on iOS.
On Mac I haven't seen any difference, still OBS studio sees the window as double size and needs to be resized by hand, but mine is a very unusual scenario.

**Reply by ajwfrost** [reply]
`requestedDisplayResolution` seems to be used in a couple of places:
* under the `<initialWindow>` tag, this will result in a Mac desktop application being packaged up with the `NSHighResolutionCapable` key being set to true/false depending on whether that value is high/standard respectively
* under the `<initialWindow>` tag, on Windows this means that the application will notify the OS that it can support high-res scaling so the scaling value of the monitor will be accommodated appropriately
* under the `<iPhone>` tag, this results in an iOS application being able to scale for retina screens so essentially for a high-res display the AIR display scales up x2

So if you've got a Windows 4K monitor and have this set as "high", if you've got your "scale and layout" value of the display settings set at something above 100%, you should see the AIR window also increase in size accordingly..

thanks

**Reply by FliplineStudios** [reply]
> * under the `<initialWindow>` tag, on Windows this means that the application will notify the OS that it can support high-res scaling so the scaling value of the monitor will be accommodated appropriately
> 
> So if you've got a Windows 4K monitor and have this set as "high", if you've got your "scale and layout" value of the display settings set at something above 100%, you should see the AIR window also increase in size accordingly..
>

Strange, I'm seeing that scaling behavior whether I have it set to "high" or "standard" on Windows 11, or even if I remove the requestedDisplayResolution tag in initialWindow.  On a 4K monitor at 150% scale, the full-screen AIR game runs at 2560x1440 regardless of this setting (and at full 4K when set to 100% scale.)  Unless Windows 11 handles this scaling naturally and it's earlier versions of Windows that would need this?  Either way, I guess I'll leave it in and set it to "high" just in case :)

**Reply by ajwfrost** [reply]
I believe that if AIR doesn't give the OS information about its scaling capabilities, then Windows will handle the scaling instead, so you'd likely see roughly the same -> but if Windows does it, some of the text etc may appear a little blurred?

**Reply by FliplineStudios** [reply]
Oh you're right, that's exactly it, both the text and the graphics are blurrier when set to "standard" when the display scale is changed.  I tried it on a 1080p monitor and could really see the difference there, it was harder to notice on 4K.  Thanks for the insight!

**Reply by shaucky** [reply]
On Macintosh, if the `<requestedDisplayResolution>` is set `standard`, the AIR application will be very blurry, but when it is set `high`, it would be clear as it is testing. On Windows, the behavior is similar with Macintosh.

I think it's important for Macintosh to set `<requestedDisplayResolution>` with `high`, but not on Windows if the font is pixelated.
